react/components/RenderProvider.tsx

```diff
@@ -654,6 +654,19 @@ export class RenderProvider extends Component<
     }
   }
 
+  private mergeRouteParams<T extends { params: any }>(
+    matchingPage: T,
+    transientRoute: { params: any }
```

Comment:
Why this property `transientRoute` isn't of type `T`?

Reply:
Because typing it as T makes the function expect that `matchingPage` and `transientRoute` have the same type, which doesn't always apply.